src/Middleware/Rewrite/src/IISUrlRewrite/ServerVariables.cs

```diff
@@ -46,6 +46,9 @@ public static PatternSegment FindServerVariable(string serverVariable, ParserCon
                 case "HTTP_HOST":
                     managedVariableThunk = () => new HeaderSegment(HeaderNames.Host);
                     break;
+                case "HTTP_HOST_NAME":
```

Comment:
Is this a new value being invented in this component or one that IIS URL Rewrite already supported? I don't see a reference to this server variable anywhere.

Reply:
This seems to be a new value. I'm up in the air whether to add this. This behavior has existed since 1.1, but it may be worth the breaking change to remove the port. Not really sure what is best.

Reply:
> but it may be worth the breaking change to remove the port. Not really sure what is best.

I.e. change the existing `HTTP_HOST` value? That value is described as being the value of the `Host` header. If that includes the port, then we should leave it alone. If it doesn't include the port in IIS (but we **did** include it in our version) then it makes sense to fix this.

As for adding a new value here, I'm not necessarily opposed, it just means we're kinda moving this beyond just a system to allow "compatibility" with previous rules. Now we're adding new functionality that the system the user is migrating from never actually had.

Reply:
@anurse 
> If that includes the port, then we should leave it alone. If it doesn't include the port in IIS (but we did include it in our version) then it makes sense to fix this.
I can understand, #9713 says `host` includes port number, how we can go over with this?

Reply:
Right, I was saying that if the IIS version of `HTTP_HOST` includes the port, so should we. So that's clear. We should not change `HTTP_HOST` (and your PR doesn't do that). I just wanted to understand if we could justify changing the current value.

Since we can't, as you said, adding a new variable (`HTTP_HOST_NAME`) is our only option. That's OK with me, but as I said, it does mean that we are no longer just providing a compatibility layer for existing IIS URL Rewrite rules. We're now introducing a **new** variable that doesn't exist in IIS, so rules that use that variable may no longer be compatible with IIS.